### ISSUE TITLE
Midterm report Peer review

### ISSUE THREAD

**Issue by ybenso** [Austin-Li-1123/Tesla_forcasting_feature_analysis#5]
Tesla Forecasting Feature Analysis

Summary:
The aim of the project is to build a model able to forecast Tesla stock prices, based on previous prices of the stock and similar companies, and tweets.

What I liked:
The project is really interesting, trying to connect numerical and text data. Tesla is really a fascinating company, and its stock price is often said to follow what Elon Musk is tweeting. 
There is a good feature engineering approach for choosing which polynomial features to consider. The best degree in different polynomial transformations is found with the MSE on train and test data, and we can clearly see when the model starts overfitting thanks to the plot.
They warn about the fact that they may use another loss function to test the effectiveness of the model, which is also a good point to consider with this specific type of data.
There is also a clear view of what is remaining to be done. There is a good consciousness of choosing the most relevant features, with PCA.

Room for improvement:
I think it is also possible to consider an ARIMA approach to model this time series, which can be combined with the tweets data. 
They also want to compare how the model performs during 2008 and 2020, but I think it might be difficult to do that if the same kind of crisis is not present between 2019 and 2021.
I think the way that tweets and news are going to be collected is still unclear, but there is still room to find out how to do that. It might be interesting to consider an embedding of each tweet, to capture the meaning of the tweet (a vectorizer such as word2vec or Bert maybe?).
